### PR TITLE
Task #6- Async (optional)

### PR DESCRIPTION
**Pull request by FerLanzaC**
# PR name: Task #6- Async (optional)
-- 

## Description: 
Solve this CodeWars Kata test using your asynchronous abilities.

- Write an async function which takes an apiUrl and jokeId which returns a promise.
- The data will need to be filtered to get the specified joke by id.
- When you got the joke it should be accessible through a simple API of saySetup and sayPunchLine methods.
- Handle error cases
- If a joke can't be found throw an error message in this format new Error('No jokes found id: {jokeId}').
- Getting jokes from a another API URL may return a different data shape, throw this error message new Error('No jokes at url: {url}') for an unexpected shape.
- Throw error in a promise style
- Get the data using the mocked fetch(url) function, which implements the basics of the fetch api.

PR name match with the ticket respective in Trello. 
  
## Type of change:  
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested?
Use this link and copy the code in the solution.
https://www.codewars.com/kata/5a353a478f27f244a1000076/train/javascript

### CODE REVIEW

File: method.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+async function sayJoke(apiUrl, jokeId){
+  const response = await fetch (apiUrl)
```
Comment (apadillaAK) [inline method.js:2]:
```suggestion
  const response = await fetch(apiUrl)
```

File: method.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  const json = await response.json()
+  for(let joke of json.jokes){
+      if (joke.id == jokeId){
+        return{saySetup: () => joke.setup,
```
Comment (apadillaAK) [inline method.js:6]:
```suggestion
        return {saySetup: () => joke.setup,
```

File: method.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+               sayPunchLine: () => joke.punchLine
+              };
+   }
+    throw new Error (`No jokes found id: ${jokeId}`);
+  }).catch(error => {
+    throw (error instanceof TypeError)? new Error (`No jokes at url: ${apiUrl}`):error
```
Comment (apadillaAK) [inline method.js:12]:
```suggestion
    throw (error instanceof TypeError) ? new Error(`No jokes at url: ${apiUrl}`) : error
```

File: method.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  const response = await fetch(apiUrl)
+  const json = await response.json()
+  for(let joke of json.jokes){
+      if (joke.id == jokeId){
+        return {saySetup: () => joke.setup,
+               sayPunchLine: () => joke.punchLine
+              };
+   }
+    throw new Error (`No jokes found id: ${jokeId}`);
+  }).catch(error => {
```
Comment (apadillaAK) [inline method.js:11]: I didn't find any `try` block, to have this `catch` here

Comment (FerLanzaC) [reply method.js]: Done

File: task06-async-optional/method.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+async function sayJoke(apiUrl, jokeId) {
+  const response = await fetch(apiUrl)
+  const json = await response.json()
+  const jokesChristmas = await json.jokes;
+  if (!jokesChristmas) {
+      throw new Error (`No jokes at url: ${apiUrl}`);
+  } else {
+    for(let joke of json.jokes) {
```
Comment (apadillaAK) [inline task06-async-optional/method.js:8]:
```suggestion
    for(let joke of jokesChristmas) {
```

**Review by apadillaAK — CHANGES_REQUESTED** [review]

**Comment by FerLanzaC** [comment]
Done!

**Pull request merged**